lib/mdspell/spell_checker.rb

```diff
@@ -29,10 +29,11 @@ def initialize(filename)
     # Returns found spelling errors.
     def typos
       results = []
-
+      ignored = Regexp.new(Configuration[:ignored].join('|'),  Regexp::EXTENDED | Regexp::IGNORECASE) unless Configuration[:ignored].empty?
```

Comment:
This is the 'magic' here:

- Join all expressions together with a pipe character, creating a union of all words / expressions with a logical 'OR'.
- Compile it into a single regular expression (build a state machine, so that the match on line 36 is extremely efficient as the regex is pre-compiled), ignore cases because why would we care about cases, and offer extended regex because why not.
- Do this outside of the loop, so it can be shared for all words